### PR TITLE
Issue 32 Resolution - Range to Distance switch

### LINKED ISSUE #32: Name Duplication: Dynamic.Mission.DISTANCE & Dynamic.Mission.RANGE
### Desired capability or behavior.

`Dynamic.Mission.RANGE` and `Dynamic.Mission.DISTANCE` have the same meaning in Aviary.
There are also rates associated with both values.
The difference is probably a holdover from the LEAPS2 and GASPy codes.
We only need one of them.
We should choose one and rename throughout Aviary.
The consolidated standard will be `Dynamic.Mission.DISTANCE`.

### Is your feature request related to a problem? Please describe.

_No response_

### Associated Bug Report

_No response_

### PR DESCRIPTION
**Pull request by ehariton**
### Summary

`Dynamic.Mission.RANGE` was removed and replaced with `Dynamic.Mission.DISTANCE`.

To implement this, some additional names were replaced:

- `initial_range` -> `initial_distance`
- `states:range` -> `states:distance`
- `_Dynamic.Mission` was removed from all files and longer but easier to find names were introduced.
- Imported `Dynamic` to test_FLOPS_based_sizing_N3CC.py to enable it to pass tests
- `"distance_rate"`, `Dynamic.Mission.RANGE_RATE` -> `Dynamic.Mission.DISTANCE_RATE`
- `.RANGE_RATE` - > `.DISTANCE_RATE` 
- (note there is still a file called range_rate which has a RangeRate() function which should eventually be renamed to DistanceRate()
- `"range"`, `'range'`, `"distance"`, `'distance'`, `Dynamic.Mission.RANGE` -> `Dynamic.Mission.DISTANCE`
- `Dynamic.Mission.RANGE`, `Dynamic.Mission.RANGE_RATE`, `RANGE`, & `RANGE_RATE` were removed from variables.py
- Old names of range_rate and range were retained in the comments in variable_meta_data.py

### Related Issues

- Resolves #32 

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Review by johnjasa — APPROVED** [review]
Thanks for the detail-oriented work here, Eliot! I dig it.

File: aviary/docs/getting_started/onboarding_level3.ipynb [hunk 1]
```diff
@@ -340,13 +340,13 @@
...
-    "            \"takeoff_range_con=takeoff_range-climb_start_range\",\n",
+    "            \"takeoff_distance_con=takeoff_range-climb_start_range\",\n",
     "            \"takeoff_vel_con=takeoff_vel-climb_start_vel\",\n",
     "            \"takeoff_alt_con=takeoff_alt-climb_start_alt\"\n",
     "        ],\n",
     "        takeoff_mass_con={'units': 'lbm'}, takeoff_mass={'units': 'lbm'},\n",
     "        climb_start_mass={'units': 'lbm'},\n",
-    "        takeoff_range_con={'units': 'ft'}, takeoff_range={'units': 'ft'},\n",
+    "        takeoff_distance_con={'units': 'ft'}, takeoff_range={'units': 'ft'},\n",
     "        climb_start_range={'units': 'ft'},\n",
```
Comment (Kenneth-T-Moore) [inline aviary/docs/getting_started/onboarding_level3.ipynb:350]: I noticed there is a "climb_start_range" that should probably also be changed to distance.

File: aviary/mission/flops_based/phases/detailed_landing_phases.py [hunk 2]
```diff
@@ -37,10 +37,7 @@
     InitialGuessState, PhaseBuilderBase)
 from aviary.utils.aviary_values import AviaryValues
 from aviary.variable_info.functions import setup_trajectory_params
-from aviary.variable_info.variables import Dynamic as _Dynamic
-from aviary.variable_info.variables import Mission
-
-Dynamic = _Dynamic.Mission
+from aviary.variable_info.variables import Dynamic, Mission
```
Comment (Kenneth-T-Moore) [inline aviary/mission/flops_based/phases/detailed_landing_phases.py:40]: All of these detailed landing phases have a "max_range" which should become "max_distance".

File: aviary/mission/flops_based/phases/detailed_takeoff_phases.py [hunk 3]
```diff
@@ -51,10 +51,7 @@
     InitialGuessState, InitialGuessTime, PhaseBuilderBase)
 from aviary.utils.aviary_values import AviaryValues
 from aviary.variable_info.functions import setup_trajectory_params
-from aviary.variable_info.variables import Dynamic as _Dynamic
-from aviary.variable_info.variables import Mission
-
-Dynamic = _Dynamic.Mission
+from aviary.variable_info.variables import Dynamic, Mission
```
Comment (Kenneth-T-Moore) [inline aviary/mission/flops_based/phases/detailed_takeoff_phases.py:54]: All of these detailed takeoff phases have a "max_range" which should become "max_distance".

File: aviary/docs/getting_started/onboarding_level3.ipynb (hunk 1 above)
Comment (ehariton) [reply aviary/docs/getting_started/onboarding_level3.ipynb:350]: for sure there are a bunch of other references to `range` in the code. I hesitate to do a global find/replace as I'm sure that will brick things. There are around 21 instances of `climb_start_range` I'll do a quick find/replace on those and see if all the tests are still working afterwards.

File: aviary/mission/flops_based/phases/detailed_landing_phases.py (hunk 2 above)
Comment (ehariton) [reply aviary/mission/flops_based/phases/detailed_landing_phases.py:40]: John was hoping to name distance_max as range I think. 
we should never be using a naming scheme with `max` at the start.

**Review by Kenneth-T-Moore — CHANGES_REQUESTED** [review]
This looked pretty good. Just noted a few more variables with "range" in the name that need to be changed over to distance.

**Comment by ehariton** [comment]
Made updates based on Ken's comments, tests passing locally, pushing up.

**Review by Kenneth-T-Moore — APPROVED** [review]

**Pull request merged**